doc/sphinx/source/howto/derive-for-training.rst

```diff
@@ -0,0 +1,140 @@
+.. derive-for-training.rst
+
+#########################
+Derive a trainable model 
+#########################
+
+Documentation in this section describes one of the ways to derive a trainable 
+model from an inference model.
+
+We can derive a trainable model from any graph that has been constructed with 
+weight-based updates. For this example named ``mnist_mlp.cpp``, we start with a hand-designed inference model and convert it to a model that can be trained 
+with nGraph. 
+
+Additionally, and to provide a more complete walk-through that *also* trains the 
+"trainable" model, our example includes the use of a simple data loader for the 
+MNIST data.
+
+* :ref:`model_overview`
+* :ref:`code_structure`
+  - :ref:`inference`
+  - :ref:`loss`
+  - :ref:`backprop`
+  - :ref:`update`
+
+
+.. _model_overview:
+
+Model overview
+==============
+
+The nGraph API was designed for automatic graph construction under direction of 
+a framework. Without a framework, the process is somewhat tedious, so the example 
```

Comment:
Maybe we should be a bit less self deprecating here. :)  Something like "The graph construction API is lower-level than a typical framework, so writing a model in it directly is somewhat akin to programming in assembly language. Because of this, the example selected is..."